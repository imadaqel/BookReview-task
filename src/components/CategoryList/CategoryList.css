.section {
    margin: 0;
    padding: 0;
}
.section-content {
    max-width: 800px;
    margin: auto;
    padding: 5rem 0;
}

.section-content > h1 {
    margin: 0;
    padding: 0;
}


.rec.rec-arrow.rec-arrow-left {
    position: absolute;
    left: 20rem;
    border-radius: 0
}
.rec.rec-arrow.rec-arrow-right {
    position: absolute;
    right: 20rem;
    border-radius: 0
}
.rec.rec-slider-container {
    margin: 0;
}
.rec.rec-carousel-wrapper {
    margin: 1rem 0 0;
}
.card {
    display: flex;
    flex-direction: column;
    width: 80%;
}
.card  > h4 {
    margin: 0;
    font-size: 14px;
}
.card a > p {
    margin: 0;
    font-size: x-small
}
.card a > img  {
    width: 13rem;
    height: 16rem;
    width: 100%;
}
.section-content > h1 {
    padding: 0 1.2rem;
}
@media only screen and (max-width: 1024px){
    .rec.rec-arrow.rec-arrow-left {
        left: 5rem;
    }
    .rec.rec-arrow.rec-arrow-right {
        right: 5rem;
    } 
    .section-content > h1 {
       text-align: center;
    }
}
@media only screen and (max-width: 768px) and (min-width: 425px) {
    .nonfiction-header {
        text-align: center;
        margin-left: 0;
        padding-top: 2rem;
    }
    .section-content{
        padding: 4rem 7rem;
    }    
    .card {
    width: 80%;   
    }
    .rec.rec-arrow.rec-arrow-left {
        left: 2rem;
    }
    .rec.rec-arrow.rec-arrow-right {
        right: 2rem;
    }
} 
@media only screen and (max-width: 425px) and (min-width: 375px)  {
    .card {
        width: 50%;
    }
}
@media only screen and (max-width: 375px) {
    .card {
        width: 40%;
    }
    .rec.rec-arrow.rec-arrow-left {
        left: 3rem;
    }
    .rec.rec-arrow.rec-arrow-right {
        right: 3rem;
    }
}