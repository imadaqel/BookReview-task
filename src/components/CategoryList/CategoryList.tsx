import React from 'react'
import './CategoryList.css'
import Carousel from 'react-elastic-carousel';
import {Link} from 'react-router-dom';

export default function CategoryList(props:any) {

    interface IbreakPoints {
        width: number;
        itemsToShow: number;
    }
    interface IItem {
        id: number;
        image: string;
        title: string;
        author: string;
        reviewedBy: string;
    }
    const title:string = props.title;
    const bgColor:string = props.bgColor;
    const titleColor:string = props.titleColor;
    const pColor:string = props.pColor;

    const items = props.items;
    const breakPoints: IbreakPoints[] = [
        {width: 200, itemsToShow: 1},
        {width: 430, itemsToShow: 3},
        {width: 800, itemsToShow: 5}
    ]
    return (
        <section className={"section"} style={{backgroundColor: bgColor}}>
            <section className="section-content">
                <h1 style={{color: titleColor}}>{title}</h1>
                <Carousel breakPoints={breakPoints} isRTL={false} pagination={false}>
                    {items && items.length > 0 && items.map((item: IItem) =>
                        <div className="card" key={item.id}>
                            <Link to={`/book/${item.id}`}>
                                <img src={item.image} alt={`can't load ${item.title}`}/>
                                <h4 style={{color: titleColor}} >{item.title}</h4>
                                <p style={{color: pColor}}>BY {item.author}</p>
                                <p style={{color: pColor}}>Reviewed By {item.reviewedBy}</p>
                            </Link>
                        </div>
                    )}
                </Carousel>
            </section>
        </section>
    )
}
