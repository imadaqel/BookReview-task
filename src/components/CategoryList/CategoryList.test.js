import React from 'react';
import renderer from 'react-test-renderer';
import CategoryList from './CategoryList';

it('renders CategoryList correctly when there are no items', () => {
  const tree = renderer.create(<CategoryList />).toJSON();
  expect(tree).toMatchSnapshot();
});

