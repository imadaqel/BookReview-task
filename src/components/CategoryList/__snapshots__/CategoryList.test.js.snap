// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CategoryList correctly when there are no items 1`] = `
<section
  className="section"
  style={
    Object {
      "backgroundColor": undefined,
    }
  }
>
  <section
    className="section-content"
  >
    <h1
      style={
        Object {
          "color": undefined,
        }
      }
    />
    <div
      className="sc-EHOje eOvqeq rec rec-carousel-wrapper "
      style={Object {}}
    >
      <div
        className="sc-ifAKCX kQtsWC rec rec-carousel"
        size={
          Object {
            "height": 0,
          }
        }
        style={
          Object {
            "height": 0,
          }
        }
      >
        <button
          className="sc-bdVaJa dsyKYi rec rec-arrow rec rec-arrow-left"
          disabled={true}
          onClick={[Function]}
          style={Object {}}
          tabIndex={0}
          type="button"
        >
          ❮
        </button>
        <div
          className="sc-htpNat iDfeOS rec rec-slider-container"
        >
          <div
            className="sc-bxivhb ktZOgd rec rec-slider"
            style={
              Object {
                "left": "0px",
                "right": "auto",
                "top": "auto",
                "transition": "all 500ms ease",
              }
            }
          >
            <div
              className="rec rec-swipable"
              onMouseDown={[Function]}
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "row",
                }
              }
            />
          </div>
        </div>
        <button
          className="sc-bdVaJa dsyKYi rec rec-arrow rec rec-arrow-right"
          disabled={true}
          onClick={[Function]}
          style={Object {}}
          tabIndex={0}
          type="button"
        >
          ❯
        </button>
      </div>
    </div>
  </section>
</section>
`;

exports[`renders correctly when there are no items 1`] = `
<section
  className="section"
  style={
    Object {
      "backgroundColor": undefined,
    }
  }
>
  <section
    className="section-content"
  >
    <h1
      style={
        Object {
          "color": undefined,
        }
      }
    />
    <div
      className="sc-EHOje eOvqeq rec rec-carousel-wrapper "
      style={Object {}}
    >
      <div
        className="sc-ifAKCX kQtsWC rec rec-carousel"
        size={
          Object {
            "height": 0,
          }
        }
        style={
          Object {
            "height": 0,
          }
        }
      >
        <button
          className="sc-bdVaJa dsyKYi rec rec-arrow rec rec-arrow-left"
          disabled={true}
          onClick={[Function]}
          style={Object {}}
          tabIndex={0}
          type="button"
        >
          ❮
        </button>
        <div
          className="sc-htpNat iDfeOS rec rec-slider-container"
        >
          <div
            className="sc-bxivhb ktZOgd rec rec-slider"
            style={
              Object {
                "left": "0px",
                "right": "auto",
                "top": "auto",
                "transition": "all 500ms ease",
              }
            }
          >
            <div
              className="rec rec-swipable"
              onMouseDown={[Function]}
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "row",
                }
              }
            />
          </div>
        </div>
        <button
          className="sc-bdVaJa dsyKYi rec rec-arrow rec rec-arrow-right"
          disabled={true}
          onClick={[Function]}
          style={Object {}}
          tabIndex={0}
          type="button"
        >
          ❯
        </button>
      </div>
    </div>
  </section>
</section>
`;
